File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.kt
class Solution {
    fun longestConsecutive(nums: IntArray): Int {
        val lookup = nums.toSet()

        var longest = 0
        for (num in nums) {
            if ((num - 1) !in lookup) {
                var current = 1
                while ((num + current) in lookup) {
                    current++
                }
                longest = maxOf(longest, current)
            }
        }

        return longest
    }
}

File: Data Structures & Algorithms/majority-element/submission-2.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        if len(nums) < 2:
            return nums[0]

        majority = nums[0]
        current = 1

        for num in nums[1:]:
            if num == majority:
                current += 1
            else:
                current -= 1

                if current == 0:
                    majority = num
                    current = 1
        
        return majority